### PR TITLE
Add test for vfat formatted config drive

### LINKED ISSUE #1643: add test for vfat cloud-init drive
and fix it if needed - extracted from "cloud-init from config drive not working in v0.8.0 (regression) #1614"

### PR DESCRIPTION
**Pull request by rovjuvano**
PR to resolve #1643

The new test requires rancher/os-kernel#51 to pass.

### CODE REVIEW

File: assets/scripts_ssh_config [hunk 1]
```diff
@@ -1,3 +1,4 @@
 StrictHostKeyChecking no
 UserKnownHostsFile /dev/null
+PasswordAuthentication no
```
Comment (SvenDowideit) [inline assets/scripts_ssh_config]: this is unrelated - and some users do use passwords :/

**Comment by SvenDowideit** [comment]
nice!

File: assets/scripts_ssh_config (hunk 1 above)
Comment (rovjuvano) [inline assets/scripts_ssh_config]: There may be a more isolated way to do this, but the test hangs in WaitForSSH at a password prompt without it.

Comment (SvenDowideit) [inline assets/scripts_ssh_config]: mmm, I think that means the config drive isn't working right - the ssh key should get installed, and that's how the WaitForSSH gets in.

kinda very weird that it works for the iso version of this test.

Comment (rovjuvano) [inline assets/scripts_ssh_config]: You're right. When the config drive fails it causes the entire suite to never finish. I can see the point that this is an unrelated issue, but until the test runs with a kernel that includes iso8859-1 support, it will fail. I haven't have a chance to figure out how to do that.

Comment (SvenDowideit) [inline assets/scripts_ssh_config]: oh duh - good point! I'm building 4.9.31 with your PR atm, and once its published, I'll update mater to use it too - after that, rebasing will re-run the tests with your fix.

**Comment by SvenDowideit** [comment]
@rovjuvano can you please remove the ssh_config change, rebase to the new master and squash?

**Comment by rovjuvano** [comment]
The 4.9.31 os-kernel build is still missing the nls_iso8859-1 module. When I build it locally, it does. Not sure what's going on here, but I noticed quite a few modules listed in modules.list that are in extras instead of the main tarball.

**Comment by SvenDowideit** [comment]
ah sorry, i think I built the wrong things from the wrong place. rebuilding 4.9.31 atm.

that tarball generateion code is pretty weird.

**Comment by SvenDowideit** [comment]
and success:

```
[rancher@rancher-dev ~]$ 
[rancher@rancher-dev ~]$ ls -la
total 12
drwxr-xr-x    3 rancher  rancher       4096 Jun  9 03:22 .
drwxr-xr-x    4 root     root          4096 Jun  9 03:21 ..
drwx------    2 rancher  rancher       4096 Jun  9 03:22 .ssh
[rancher@rancher-dev ~]$ sudo blkid
/dev/sda: SEC_TYPE="msdos" LABEL="config-2" UUID="43DB-BB5D" TYPE="vfat"
/dev/vda: LABEL="RANCHER_STATE" UUID="18e624b6-bd7a-495e-9385-edcea3eea8e2" TYPE="ext4"
[rancher@rancher-dev ~]$ uname -a
Linux rancher-dev 4.9.31-rancher #1 SMP Fri Jun 9 03:06:05 UTC 2017 x86_64 GNU/Linux
```

**Comment by SvenDowideit** [comment]
@rovjuvano sorry, I've been distracted :/

atm, Drone is failing because you need to `go fmt tests/cloud_init_test.go`

**Comment by SvenDowideit** [comment]
ok, I've added the go fmt in #1940 - closing this one

thank you @rovjuvano for doing this!
